src/parser/name_resolution.c

```diff
@@ -9187,6 +9194,17 @@ pt_resolve_names (PARSER_CONTEXT * parser, PT_NODE * statement, SEMANTIC_CHK_INF
 			      node->info.name.original);
 		  return NULL;
 		}
+
+	      for (entity = spec->info.spec.flat_entity_list; entity; entity = entity->next)
+		{
+		  db = entity->info.name.db_object;
+		  if (db_is_system_class (db) && (db_check_authorization (db, DB_AUTH_UPDATE) != NO_ERROR))
```

Comment:
There is the same code on line 9222.
Do both conditions have to be met? Isn't this an OR condition?